graph/db/sql_store.go

```diff
@@ -325,6 +325,39 @@ func (s *SQLStore) HasNode(ctx context.Context,
 	return lastUpdate, exists, nil
 }
 
+// HasNode determines if the graph has a vertex identified by the
+// target node identity public key.
+//
+// NOTE: part of the Store interface.
+func (s *SQLStore) HasNode(ctx context.Context, pubKey [33]byte) (bool, error) {
+	var (
+		v      = lnwire.GossipVersion1
+		exists bool
+	)
+	err := s.db.ExecTx(ctx, sqldb.ReadTxOpt(), func(db SQLQueries) error {
+		_, err := db.GetNodeByPubKey(
```

Comment:
nit: in sql this could be a cheap query returning 0/1 depending on whether the node exists.